### DISCUSSION TITLE
emit field value change on component created

### DISCUSSION THREAD

**Discussion by mehrancodes** [laravel/nova-issues#4300] [Q&A] [answered]
This week I was working on upgrading a third-party field to get compatible with Nova 4 and accidentally saw that Nova SelectField and so many other fields are doing emit on value change, which is super powerful on the third-party components to conditionally do something if a field value got changed.

But also it would be so much great if we could use `emitFieldValueChange` when on the fields component created cycle. this way on the third-parties we can initial the our component's value conditionally on update page.

at the moment only SelectField supports value change on created cycle and when value gets changed.

**Comment by crynobone** [comment, marked as answer]
> But also it would be so much great if we could use emitFieldValueChange when on the fields component created cycle.

No, this would clash with **Dependent Fields** feature.

> at the moment only SelectField supports value change on created cycle and when value gets changed.

All other fields emits the event from either `DependentFormField` or `FormField` mixins.
